client/components/map-component.jsx

```diff
@@ -36,50 +44,85 @@ export default class MapComponent extends React.Component {
     this.autocomplete = autocomplete;
   }
 
-  onPlaceChanged() {
+  onPlaceChanged(e) {
     if (this.autocomplete !== null) {
       const place = this.autocomplete.getPlace();
       const lat = place.geometry.location.lat();
       const lng = place.geometry.location.lng();
-      this.setState({ markerPosition: { lat, lng } });
+      const name = place.name;
+      this.setState({ markerPosition: { lat, lng }, name });
     }
   }
 
+  nullValue(e) {
+    if (e.target.value === '') {
+      this.setState({ markerPosition: null, name: null });
+    }
+  }
+
+  showLogModal(e) {
+    if (e.target.name === 'logModal' && !this.state.logModal) {
+      this.setState({ logModal: true });
+    } else if (e.target.getAttribute('name') === 'cancel' && this.state.logModal) {
+      this.setState({ logModal: false });
+    }
+  }
+
+  hideLogModal() {
+    this.setState({ logModal: false });
+  }
+
   render() {
-    const myLatLng = {
-      lat: this.state.userLat,
-      lng: this.state.userLong
-    };
+    let myLatLng;
+    if (this.state.markerPosition) {
+      myLatLng = this.state.markerPosition;
+    } else {
+      myLatLng = {
+        lat: this.state.userLat,
+        lng: this.state.userLong
+      };
+    }
+    const { markerPosition, name, logModal } = this.state;
+    const { showLogModal, hideLogModal } = this;
+    const contextValue = { markerPosition, name, logModal, showLogModal, hideLogModal };
     return (
-      // process.env.GOOGLE_MAPS_API_KEY, libraries={['places']}
-      <LoadScript
-        googleMapsApiKey={process.env.GOOGLE_MAPS_API_KEY}
-        libraries={libraries}
-      >
-
-        <GoogleMap
-          mapContainerStyle={containerStyle}
-          center={myLatLng}
-          zoom={12}
-          options={options}
-        >
-          <Autocomplete
-            onLoad={this.onLoad}
-            onPlaceChanged={this.onPlaceChanged}
+      // process.env.GOOGLE_MAPS_API_KEY, libraries={['places']}    fakekey
+      <AppContext.Provider value={contextValue}>
+        <>
+          <LoadScript
+            googleMapsApiKey={fakekey}
```

Comment:
Use `process.env.GOOGLE_MAPS_API_KEY` instead of a fake value like this. We should not need to keep changing the code to protect our keys.